src/internal/connector/graph/service.go

```diff
@@ -307,9 +307,16 @@ func (handler *LoggingMiddleware) Intercept(
 		// special case for supportability: log all throttling cases.
 		if resp.StatusCode == http.StatusTooManyRequests {
 			logger.Ctx(ctx).Infow("graph api throttling", "method", req.Method, "url", req.URL)
-		}
-
-		if resp.StatusCode != http.StatusTooManyRequests && (resp.StatusCode/100) != 2 {
+		} else if resp.StatusCode/100 == 4 {
+			respDump, _ := httputil.DumpResponse(resp, true)
+			logger.Ctx(ctx).Infow(
+				"graph api error",
+				"status", resp.Status,
+				"method", req.Method,
+				"url", req.URL,
+				"response", string(respDump),
+			)
+		} else if resp.StatusCode/100 != 2 {
```

Comment:
What is this solving that the above conditional (toggling all api logging) doesn't?

Reply:
This was done primarily to make sure we get some extra information about why requests failed in ark without having to enable debug logging which can be really noisy. The primary status code that I think logging resopnse could help is just 400 (not necessarily 4xx) as having the response might help us figure out what is causing a failure like in the case of possible "restricted directories". Let me know what you think.

Reply:
mm, I still think we'll get all this information out of the new structured errors.  But don't let me stop you.

If we're only looking for 400, we should explicitly catch that; not all 4xx.
```
else if resp.Status == http.StatusBadRequest {
```